From: Marc Zyngier <marc.zyngier@arm.com>
Date: Mon, 13 Nov 2017 16:21:33 +0000
Subject: irqchip/gic-v4: Add forward definition of struct irq_domain_ops
Patch-mainline: v4.15-rc1
Git-commit: 47f9d0bf526058b3fdc077698fcb19748d5700e4
References: FATE#323954

In some randconfig scenarios, including arm-gic-v4.h results
in a spurious wawrning about the $SUBJECT structure not being
defined. Adding a forward definition keeps it quiet.

Reported-by: Arnd Bergmann <arnd@arndb.de>
Signed-off-by: Marc Zyngier <marc.zyngier@arm.com>
Acked-by: Alexander Graf <agraf@suse.de>
---
 include/linux/irqchip/arm-gic-v4.h |    1 +
 1 file changed, 1 insertion(+)

--- a/include/linux/irqchip/arm-gic-v4.h
+++ b/include/linux/irqchip/arm-gic-v4.h
@@ -109,6 +109,7 @@
 int its_unmap_vlpi(int irq);
 int its_prop_update_vlpi(int irq, u8 config, bool inv);
 
+struct irq_domain_ops;
 int its_init_v4(struct irq_domain *domain, const struct irq_domain_ops *ops);
 
 #endif
